util/shop/shop.js

```diff
@@ -125,11 +125,18 @@ const royalty = (userId, callback) => {
   return false;
 };
 
-const buy = (userId, item, quantity, callback, errorCallback) => {
+const buy = (userId, item, quant, callback, errorCallback) => {
   if (utils.hasProperty(enabledPowerups, item)) {
     let { price } = enabledPowerups[item];
-    if (!Number.isNaN(Number.parseInt(quantity, 10)) && enabledPowerups[item].quantified) {
-      price *= Number.parseInt(quantity, 10);
+    let quantity = 1;
+    if (quant && quant.toLowerCase() === 'max') {
+      quantity = Math.floor(data.getCoins(userId) / price);
+      if (quantity === 0) quantity = 1;
+    } else if (!Number.isNaN(Number.parseInt(quant, 10))) {
+      quantity = Number.parseInt(quant, 10);
+    }
+    if (enabledPowerups[item].quantified) {
+      price *= quantity;
```

Comment:
can you move everything inside of `if (enabledPowerups[item].quantified)` so this code only runs when needed?

Reply:
sure thing jelly bean

Reply:
did you do the same testing after you made this change too? I know it shouldn't mess anything up but stranger things have happened

Reply:
yes